File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/sumUpTo.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/sumUpTo
 *
 * @author Alex Golub
 * @since 06-May-21, 2:10 PM
 */
class sumUpTo {
  public static double sumUpTo(int n) {
    if (n < 0) {
      throw new IllegalArgumentException();
    }
    
    return n == 0 ? 0.0 : 1.0 / n + sumUpTo(n - 1);
  }
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/sumUpTo.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/sumUpTo
 *
 * @author Alex Golub
 * @since 06-May-21, 2:10 PM
 */
function sumUpTo(n) {
  if (n < 0) throw 'n must be non-negative';
  return n === 0 ? 0 : sumUpTo(n - 1) + 1 / n;
}
